agent/agent.go

```diff
@@ -340,6 +343,19 @@ func (a *Agent) WriteOutput() (err error) {
 	return nil
 }
 
+// makeDestinationDirectory will ensure that the agent's destination directory exists. If the full
+// path exists, this is a no-op and will return nil. Otherwise, it will create any directories that do not
+// exist in the destination path.
+func (a *Agent) makeDestinationDirectory() error {
+	// MkdirAll handles cases where directories already exist, so we do not need to check for `os.ErrExist` errors.
+	if err := os.MkdirAll(a.Config.Destination, 0755); err != nil {
+		a.l.Error("Error encountered in creating destination directory", "error", err)
```

Comment:
Cool, I like it! Naming this function was tough - I think I like `ensureDirectory` better than `makeDestinationDirectory` since `make` suggests that something will absolutely be created, which won't usually be the case.